### DISCUSSION TITLE
Supported tunnels / reverse proxies

### DISCUSSION THREAD

**Discussion by MikeDev101** [MikeDev101/cloudlink#56] [General]
| Service | Works with CloudLink? |
|--------|------------------------|
| ngrok | ✅ (HTTP mode on port 3000) |
| pagekite | ✅ (TCP? or HTTP mode on port 3000) |
| pitunnel | ❌ (I've tried HTTP and TCP modes, but it doesn't work) |
| repl | ✅ (requires running the server on IP '0.0.0.0') |
| localtunnel | ✅ (mostly hit-or-miss) |
| glitch | ❌(Further evaluation needed) |
| cloudflared | ✅ (HTTP mode on port 3000) |
| heroku| ✅ ('0.0.0.0:3000') |

**Comment by DabDatBass** [comment]
how about glitch?

**Comment by MikeDev101** [comment]
> how about glitch?

Glitch lacks support for hosting servers and port forwarding. So no, at best Glitch is for applications, not for servers.

**Comment by Ancoder2009** [comment]
umm i changed the server.start(port, "localhost") to server.start(port, 10.0.0.2) and it worked

**Comment by MikeDev101** [comment]
> umm i changed the server.start(port, "localhost") to server.start(port, 10.0.0.2) and it worked

Apparently so. Just got CloudLink 3.0 running.

**Comment by MikeDev101** [comment]
Updated to add localtunnel as an option

**Comment by parakeet-live** [comment]
If I were to run this on a VPS or a dedicated server instead of my machine, could I just connect through the IP? or?

**Comment by MikeDev101** [comment]
> If I were to run this on a VPS or a dedicated server instead of my machine, could I just connect through the IP? or?

it should support a reverse proxy with HTTPS/TLS and must not complain about websockets

**Comment by parakeet-live** [comment]
Also, I tried using ngrok with cloudlink but it didn't work.

Here's what I did.
WSL client1:
python3 server_example.py
(server runs no problem)

WSL client2:
./ngrok http 3000
(server starts)

epicques:
attempt to connect, nothing.
It works on 127.0.0.1 but I'm planning on using this for one of my projects I'm working on.*

(unrelated to issue)
*I should have technically asked this earlier, but me and TheEggo (ZeroChan58) are working on osu!js. We wanna know if it's okay if we use CloudLink with our project.
https://osujs-game.wixsite.com/lazer

**Comment by qbjl** [comment]
> (unrelated to issue) *I should have technically asked this earlier, but me and TheEggo (ZeroChan58) are working on osu!js. We wanna know if it's okay if we use CloudLink with our project. https://osujs-game.wixsite.com/lazer

im going to try osu!js, but well you can use CloudLink without permission. there isn't any real holdup, this is an ordinary extension. (it's in adacraft so you know he really doesn't care who uses it.)

**Comment by ghost** [comment]
repl keeps saying that the client (me) is not masked , when on ip 0.0.0.0:3000

**Comment by MikeDev101** [comment]
added cloudflared

**Comment by MikeDev101** [comment]
converting to discussion.

**Comment by spacedouut** [comment]
what about heroku?

**Comment by MikeDev101** [comment]
> what about heroku?

Heroku works. Need to set IP to 0.0.0.0:3000. (I tested this a long time ago, so it might be different).
